diktat-ruleset/src/main/kotlin/org/cqfn/diktat/DiktatMain.kt

```diff
@@ -0,0 +1,87 @@
+/**
+ * The file contains main method
+ */
+
+package org.cqfn.diktat
+
+import org.cqfn.diktat.api.DiktatError
+import org.cqfn.diktat.api.DiktatMode
+import org.cqfn.diktat.cli.DiktatProperties
+import org.cqfn.diktat.common.utils.loggerWithKtlintConfig
+import org.cqfn.diktat.ktlint.unwrap
+import org.cqfn.diktat.ruleset.rules.DiktatRuleSetFactory
+import org.cqfn.diktat.util.tryToPath
+import org.cqfn.diktat.util.walkByGlob
+import mu.KotlinLogging
+import java.nio.file.Paths
+import kotlin.io.path.absolutePathString
+import kotlin.io.path.extension
+import kotlin.io.path.writeText
+
+@Suppress("EMPTY_BLOCK_STRUCTURE_ERROR")
+private val log = KotlinLogging.loggerWithKtlintConfig {}
+
+typealias DiktatErrorWithCorrectionInfo = Pair<DiktatError, Boolean>
+
+@Suppress(
+    "LongMethod",
+    "TOO_LONG_FUNCTION"
+)
+fun main(args: Array<String>) {
+    val properties = DiktatProperties.parse(args)
+    properties.configureLogger()
+
+    log.debug {
+        "Loading diktatRuleSet using config ${properties.config}"
+    }
+    val diktatRuleSetFactory = DiktatRuleSetFactory(properties.config)
+    val reporter = properties.reporter()
+    reporter.beforeAll()
+
+    log.debug {
+        "Resolving files by patterns: ${properties.patterns}"
+    }
+    val currentFolder = Paths.get(".")
+    properties.patterns
+        .asSequence()
+        .flatMap { pattern ->
+            pattern.tryToPath()?.let { sequenceOf(it) }
+                ?: run {
```

Comment:
Why a single function call is wrapped into a `run {}` block here?

Reply:
leftover

Reply:
removed